In [3]:
# Import the pandas library
import pandas as pd

# Read the CSV file into a DataFrame
df = pd.read_csv('./data/Project2_dataset.csv')

In [ ]:
# See the shape of the DataFrame
df.shape

(57301, 9)

This dataset contains 9 columsn with 57,301 observations. Which makes this a long data.

In [5]:
# Check columns
df.columns

Index(['Airline Name', 'Airline Country', 'Departure Airport Name',
       'Departure Airport City', 'Departure Airport Country/Region',
       'Arrival Airport Name', 'Arrival Airport City',
       'Arrival Airport Country/Region', 'Plane Name'],
      dtype='str')

This data seems to be containing information regarding a flight's information, containing information regarding its airline, the departure and arrival airports, and its name.

We can break this data down into a few datasets:
- Airline information; name and country
- Airport information; name, city, country
- Plane information; its name and the airline its operating under
- The routes that the planes fly on


In [ ]:
# Display the first few rows of the DataFrame
df.head()

,Airline Name,Airline Country,Departure Airport Name,Departure Airport City,Departure Airport Country/Region,Arrival Airport Name,Arrival Airport City,Arrival Airport Country/Region,Plane Name
0,Tuna Aero,Sweden,Norsup Airport,Norsup,Vanuatu,Bauerfield International Airport,Port-vila,Vanuatu,Harbin Yunshuji Y12;De Havilland Canada DHC-6 ...
1,Tuna Aero,Sweden,Santo Pekoa International Airport,Santo,Vanuatu,Lonorore Airport,Lonorore,Vanuatu,Harbin Yunshuji Y12;De Havilland Canada DHC-6 ...
2,Tuna Aero,Sweden,Craig Cove Airport,Craig Cove,Vanuatu,Norsup Airport,Norsup,Vanuatu,Harbin Yunshuji Y12;De Havilland Canada DHC-6 ...
3,Tuna Aero,Sweden,Bauerfield International Airport,Port-vila,Vanuatu,Norsup Airport,Norsup,Vanuatu,Harbin Yunshuji Y12;De Havilland Canada DHC-6 ...
4,Tuna Aero,Sweden,Aneityum Airport,Anelghowhat,Vanuatu,Tanna Airport,Tanna,Vanuatu,Harbin Yunshuji Y12;Pilatus Britten-Norman BN-...


## Check missing values and other anomalies

### Null values

In [7]:
# Check null values in the DataFrame
df.isnull().sum()

Airline Name                        0
Airline Country                     0
Departure Airport Name              0
Departure Airport City              0
Departure Airport Country/Region    0
Arrival Airport Name                0
Arrival Airport City                0
Arrival Airport Country/Region      0
Plane Name                          0
dtype: int64

The data seems to not have any missing values. Which is good.

### Duplicated values

In [ ]:
# Check duplicated rows in the DataFrame
df.duplicated().sum()

np.int64(0)

The data does not have any duplicated rows.

In [ ]:
# Check the number of unique values in each column
df.nunique()

Airline Name                         488
Airline Country                      106
Departure Airport Name              2784
Departure Airport City              2697
Departure Airport Country/Region     223
Arrival Airport Name                2789
Arrival Airport City                2700
Arrival Airport Country/Region       220
Plane Name                          2728
dtype: int64

## Exploring Each Column

### Countries with the most airlines

In [14]:
df['Airline Country'].value_counts()

Airline Country
United States          11207
United Kingdom          5704
Mexico                  3721
Greece                  2707
Russia                  2688
                       ...  
Uruguay                    8
Trinidad and Tobago        8
Afghanistan                6
Ireland                    5
Belarus                    4
Name: count, Length: 106, dtype: int64

### 10 Countries with the most flight coming out 

In [17]:
df['Departure Airport Country/Region'].value_counts().head(10)

Departure Airport Country/Region
United States     9366
China             7933
United Kingdom    2236
Spain             2127
Germany           2048
France            1807
Russia            1712
Italy             1679
India             1342
Brazil            1317
Name: count, dtype: int64

### 10 Countries with the most flight coming in

In [18]:
df['Arrival Airport Country/Region'].value_counts().head(10)

Arrival Airport Country/Region
United States    9262
China            7933
Canada           2414
Spain            2229
Germany          2041
France           1804
Russia           1738
Italy            1682
Brazil           1308
India            1287
Name: count, dtype: int64

### Domestic FLights

In [25]:
df[df['Departure Airport Country/Region'] == df['Arrival Airport Country/Region']].value_counts('Departure Airport Country/Region').head(10)

Departure Airport Country/Region
United States    7046
China            6778
Brazil           1101
India             918
Russia            892
Canada            646
Indonesia         607
Australia         558
Spain             548
Japan             500
Name: count, dtype: int64

### International Flight

In [29]:
df[df['Departure Airport Country/Region'] != df['Arrival Airport Country/Region']].value_counts('Departure Airport Country/Region').head(10)

Departure Airport Country/Region
United States           2320
United Kingdom          1987
Germany                 1863
Spain                   1579
France                  1325
Italy                   1275
China                   1155
Russia                   820
United Arab Emirates     643
Japan                    632
Name: count, dtype: int64

In [15]:
df.columns

Index(['Airline Name', 'Airline Country', 'Departure Airport Name',
       'Departure Airport City', 'Departure Airport Country/Region',
       'Arrival Airport Name', 'Arrival Airport City',
       'Arrival Airport Country/Region', 'Plane Name'],
      dtype='str')

## Building CSVs
CSVs that will be made from this dataset are:
- Airline
- Airport
- 